src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -718,7 +739,7 @@ public string GetSymbolAlternateName(ISymbolNode node)
             "__GCStaticRegionStart", 
             "__GCStaticRegionEnd", 
             null);
-        public ArrayOfEmbeddedDataNode ThreadStaticsRegion = new ArrayOfEmbeddedDataNode(
+        public ThreadStaticsRegionNode ThreadStaticsRegion = new ThreadStaticsRegionNode(
```

Comment:
This is placing threadstatics in a different section than they were on CoreRT before. Does this work on Linux/OSX?

Cc @sergiy-k